File: Blog/src/Blog.Dtos/Post/PostListModel.cs
using System.Collections.Generic;
using Blog.Dtos.Comment;

namespace Blog.Dtos.Post;

public class PostListModel {
    public long Id { get; set; }
    public string Photo { get; set; } = string.Empty;
    public string Header { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public List<CommentModel> Comment { get; set; } = new();
    public long CreateTime { get; set; }
    public long? UserId { get; set; }
}

File: Blog/ui/src/app/models/post-list.model.ts
import { PostTypeEnum } from "@shared/enums/post-type.enum";
import { CommentModel } from "./comment.model";

export class PostListModel {
    public id!: number;
    public photo!: string;
    public header!: string;
    public body!: string;

    public Type!: PostTypeEnum;

    public comments!: CommentModel[]

    public constructor(
        fields?: Partial<PostListModel>) {

        if (fields) {


            Object.assign(this, fields);
        }
    }
}
